Modules/Core/Common/include/itkArray2D.h

```diff
@@ -60,13 +60,33 @@ class ITK_TEMPLATE_EXPORT Array2D : public vnl_matrix<TValue>
   /** Copy-constructor. */
   Array2D(const Self & array);
 
+  /** Move-constructor.
+   * \note This move-constructor is `noexcept`, even while the move-constructor of its base class (`vnl_matrix`) is not
+   * `noexcept`, because unlike `vnl_matrix`, `Array2D` always manages its own memory. */
+  Array2D(Self && array) noexcept
+    : vnl_matrix<TValue>(std::move(array))
```

Comment:
What is the state of array after the move?

Reply:
array is an 'r-value' reference, so it should be ok as it is a temporary object.  As such it is explicitly expected to be invalid after this call.

Reply:
> What is the state of array after the move?

@blowekamp I hope that the unit tests answer this question sufficiently:
 
https://github.com/InsightSoftwareConsortium/ITK/blob/5610e233c6a6d5fc2fa9a16259c379ef41c6cdec/Modules/Core/Common/test/itkArray2DGTest.cxx#L78-L80

So after the move, the right-hand-side (the "moved-from" array) is empty.

Reply:
> As such it is explicitly expected to be invalid after this call.

[Just nitpicking] In this particular case, the moved-from object (the right-hand-side after the move) isn't really "invalid", it's just empty. In general, I think a moved-from object should still have a "valid" value. At least, it should be properly "destructible".

Reply:
So in general, I think end-users should not care too much about the state of a moved-from object. It should just be destructible!